clippy_lints/src/methods/extend_with_drain.rs

```diff
@@ -16,7 +16,10 @@ pub(super) fn check(cx: &LateContext<'_>, expr: &Expr<'_>, recv: &Expr<'_>, arg:
         //check source object
         if let ExprKind::MethodCall(src_method, _, [drain_vec, drain_arg], _) = &arg.kind;
         if src_method.ident.as_str() == "drain";
-        if let src_ty = cx.typeck_results().expr_ty(drain_vec).peel_refs();
+        let src_ty = cx.typeck_results().expr_ty(drain_vec);
+        //get immutable type 
```

Comment:
I think this comment can be left out completely, as it's implied from the context. Otherwise, it could explain a bit better what this does, as this doesn't get the immutable type but only checks it and stores this information.
```suggestion
        // check if actual src type is mutable for code suggestion
```